src/Raven.Studio/typescript/viewmodels/database/settings/databaseRecord.ts

```diff
@@ -160,6 +162,51 @@ class databaseRecord extends viewModelBase {
                 }
             });
     }
+    
+    private handleNestedSettings(dto: any) {
+        const settings = dto.Settings;
+        
+        if (!settings) {
+            return dto;
+        }
+        
+        const settingsKeys = Object.keys(settings);
+        if (settingsKeys.length === 0) {
+            return dto;
+        }
+
+        let flattenedSettings = { fs: {} };
```

Comment:
I don't see the point to detect if we should flatten or not. Just check if settings are present if yes then call flatten and this is end of story. 

```
  function flattenObject(obj, parent, res = {}){
    for(let key in obj){
        let propName = parent ? parent + '.' + key : key;
        if(typeof obj[key] == 'object'){
            flattenObj(obj[key], propName, res);
        } else {
            res[propName] = obj[key];
        }
    }
    return res;
}
```

Reply:
see: https://stackoverflow.com/questions/44134212/best-way-to-flatten-js-object-keys-and-values-to-a-single-depth-array

Reply:
done